rero_ils/alembic/64a5cc96f96e_update_claimed_issue_status.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
+# Copyright (C) 2019-2023 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Update claimed issue status."""
+
+from logging import getLogger
+
+from rero_ils.modules.items.api import Item, ItemsIndexer, ItemsSearch
+from rero_ils.modules.items.models import ItemIssueStatus
+
+# revision identifiers, used by Alembic.
+revision = '64a5cc96f96e'
+down_revision = '8d97be2c8ad6'
+branch_labels = ()
+depends_on = None
+
+LOGGER = getLogger('alembic')
+indexing_chunck_size = 1000
+
+
+def upgrade():
+    """Update item issue obsolete 'claimed' status."""
+    query = ItemsSearch() \
+        .filter('term', issue__status='claimed') \
+        .source(False)
+    uuids = []
+    for hit in query.scan():
+        record = Item.get_record(hit.meta.id)
+        LOGGER.info(f'* Processing Item#{record.pid} [{record.id}]...')
+        status = ItemIssueStatus.LATE
+        if record['issue'].get('received_date'):
+            status = ItemIssueStatus.RECEIVED
+        record['issue']['status'] = status
+        record.update(record, dbcommit=True, reindex=False)
```

Comment:
This will not work as commit is not set to `True`. The best approach is `record.update(record, dbcommit=True)` and apply `db.session.commit()` every 1000 records. You can also index the 1000 records.